### ISSUE TITLE
backup/restore: improved usability work in scope for 1.2

### ISSUE THREAD

**Issue by dianasaur323** [cockroachdb/cockroach#18822]
Tracking issue for usability work to be accomplished in 1.2 for backup / restore. In order of priority. 

- [ ] Allow restore into a non-existing database https://github.com/cockroachdb/cockroach/issues/19055
- [ ] Save credentials as cluster settings
- [x] Better compatibility with Google Cloud https://github.com/cockroachdb/cockroach/pull/19018#pullrequestreview-68083735
- [ ] Export grants in restore https://github.com/cockroachdb/cockroach/issues/18636 (stretch)

cc @danhhz @mjibson @dt @benesch @cuongdo for comments!

**Comment by cuongdo** [comment]
Could you clarify how GC compatibility should be improved? Also, in what specific ways should restoring interleaved tables be improved?

**Comment by dt** [comment]
I think the GCE fix is supporting explicit credentials, not just instance auth.

**Comment by dianasaur323** [comment]
@cuongdo yep, my understanding was that you currently have to be running within google cloud platform in order to backup / restore into Google Cloud, and I think we should have a better story of compatibility there.

For interleaved tables, my understanding is that currently, we limit what portion of the interleaved table you can restore. I'm moving outside of my comfortable zone of what I know here, so @dt will have to confirm, but my understanding is that the outer "section(?)" of the interleaved table can be restored, but the inner "section(?)" can't without also restoring the full interleaved table.

**Comment by dt** [comment]
yeah, currently it's all-or-nothing when restoring interleaved tables.

**Comment by dianasaur323** [comment]
So I removed the multiple table and interleaved table support, since our 1.2 issue list is aggressive at this point. I figured we might be able to keep these last 3 small usability nits, since they seem like they shouldn't take too long to implement. Correct me if I'm wrong?

**Comment by mjibson** [comment]
LGTM

**Comment by dianasaur323** [comment]
ps, noticed that I was missing export of grants for restore. I think we can do a better job, and since we are investing more time on restore than on import this time around, we can try to get this, and mark it as a stretch if we don't.

**Comment by dt** [comment]
we export grants, we just don't restore them, and instead treat the tables more like newly created tables (inherit from db in which they're created). this is because there were some open questions around how to handle matching users (or not) that we wanted to resolve first.

**Comment by bdarnell** [comment]
Cluster settings are visible to any user on the cluster, right? It doesn't seem like a great place to store credentials. It might be useful as a stopgap (better from an operational perspective than env vars), but we should really design a better solution for storing secrets like this. (like a `system.secrets` table, potentially with ACLs to allow non-root to run commands using secrets from the table)

**Comment by dt** [comment]
Hm. `system.settings` was supposed to be (and is) root-only, but looks like non-root can run `show cluster setting`.

**Comment by dt** [comment]
Huh, yeah, we noticed this on #17619 too, and as mentioned there, I think we should either open the table to reads or require super-user on `SHOW`. We also discussed there how to handle "sensitive" settings values (e.g. if they should not appear in event log) but that discussion was more hypothetical -- when we introduce a real case, like saved cloud credentials, we need to actually solve that.

I think we could add a `sensitive` field to Setting that causes its value to be redacted in SHOW by non-superuser and when we create the the eventlog entry.
